dialects/CMakeLists.txt

```diff
@@ -112,7 +117,11 @@ add_thorin_dialect(autodiff
         autodiff/auxiliary/autodiff_aux.h
         autodiff/auxiliary/autodiff_rewrite_inner.cpp
         autodiff/auxiliary/autodiff_rewrite_toplevel.cpp
+        autodiff/auxiliary/autodiff_mem.cpp
+        autodiff/auxiliary/autodiff_mem.h
         autodiff/normalizers.cpp
+        affine/passes/lower_for.cpp
+        mem/passes/rw/reshape.cpp
     DEPENDS
         mem
         core
```

Comment:
Can we plz keep the order here alphabetic in the name of the dialects. The reason is that the order here will be the order [here](https://anydsl.github.io/thorin2/usergroup1.html). What is more, trying to keep a dependence order is also kind of clumsy anyway. We are adding and changing dialects so it's just a matter of time till some "logical" order is screwed up anyway.